experimental/dds/tree2/src/core/tree/anchorSet.ts

```diff
@@ -402,59 +400,48 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents> {
 		}
 	}
 
+	private nextRange = 0;
+	private createEmptyDetachedField(): FieldKey {
+		const detached: DetachedField = brand(String(this.nextRange++));
+		const key = detachedFieldAsKey(detached);
+		assert(!this.root.children.has(key), 0x370 /* new range must not already exist */);
+		this.root.children.set(key, []);
+		return key;
+	}
+
 	/**
-	 * Updates paths for a range move (including re-parenting path items and updating indexes).
-	 * @param count - number of siblings to insert/delete/move.
-	 * @param srcStart - where the siblings are removed from. If undefined the operation is an insert.
-	 * @param dst - where the siblings are moved to. If undefined the operation is a delete.
-	 *
-	 * TODO:
-	 * How should anchors that become invalid, then valid again (ex: into content that was deleted, then undone) work?
-	 * Add an API to resurrect them? Store them in special detached fields? Store them in special non-detached fields?
-	 *
-	 * TODO:
-	 * How should custom anchors work (ex: ones not just tied to a specific Node)?
-	 * This design assumes they can be expressed in terms of a Node anchor + some extra stuff,
-	 * but we don't have an API for the extra stuff yet.
+	 * Decouple nodes from their parent. This method doesn't place the nodes anywhere,
+	 * so if the intention is:
+	 * 1) to remove the nodes call `removeChildren`.
+	 * 2) to move the nodes call `moveChildren`.
```

Comment:
I like how you have split this up, but I think the docs could be a little more clear. Also I like to put each sentence on its own line (this helps make line based tools like git, diffs etc) better align with out semantics.

```suggestion
	 * Decouple nodes from their parent.
	 * This removes the reference from the parent to the decoupled children, and updates the indexes of the remaining children accordingly.
	 * This does NOT update the decoupled children: both their index and parent are left at their existing values.
	 * To decouple and fixup the children, see `removeChildren` and `moveChildren`.
```